src/app/base/components/PowerTypeFields/PowerTypeFields.test.tsx

```diff
@@ -62,51 +56,38 @@ describe("PowerTypeFields", () => {
       }),
     ];
     state.general.powerTypes.data = powerTypes;
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <Formik
-          initialValues={{ power_type: PowerTypeNames.MANUAL }}
-          onSubmit={jest.fn()}
-        >
-          <PowerTypeFields />
-        </Formik>
-      </Provider>
+    renderWithMockStore(
+      <Formik
+        initialValues={{ power_type: PowerTypeNames.MANUAL }}
+        onSubmit={jest.fn()}
+      >
+        <PowerTypeFields />
+      </Formik>,
+      { state }
     );
 
     expect(
-      wrapper.find("Input[name='power_parameters.field1']").props().label
-    ).toBe("Required text");
-    expect(
-      wrapper.find("Input[name='power_parameters.field1']").props().required
-    ).toBe(true);
-    expect(
-      wrapper.find("Input[name='power_parameters.field2']").props().label
-    ).toBe("Non-required text");
+      screen.getByRole("textbox", { name: "Required text" })
+    ).toBeInTheDocument();
     expect(
-      wrapper.find("Input[name='power_parameters.field2']").props().required
-    ).toBe(false);
+      screen.getByRole("textbox", { name: "Required text" })
+    ).toBeRequired();
     expect(
-      wrapper.find("Select[name='power_parameters.field3']").props().label
-    ).toBe("Select with choices");
+      screen.getByRole("textbox", { name: "Non-required text" })
+    ).toBeInTheDocument();
     expect(
-      wrapper.find("Select[name='power_parameters.field3']").find("option")
-        .length
-    ).toBe(2);
-    expect(
-      wrapper
-        .find("Select[name='power_parameters.field3']")
-        .find("option")
-        .at(0)
-        .text()
-    ).toBe("Choice 1");
-    expect(
-      wrapper
-        .find("Select[name='power_parameters.field3']")
-        .find("option")
-        .at(1)
-        .text()
-    ).toBe("Choice 2");
+      screen.getByRole("textbox", { name: "Non-required text" })
+    ).not.toBeRequired();
+
+    const selectWithChoices = screen.getByRole("combobox", {
+      name: "Select with choices",
+    });
+    expect(selectWithChoices).toBeInTheDocument();
+    powerTypes[0].fields[2].choices.forEach((choice) => {
```

Comment:
(Optional) Would be good to work on improving readability of this, these indexes don't make it easy to understand what's the expected behaviour at a glance.